frontend/components/commons/forms/questions-form/QuestionsForm.component.vue

```diff
@@ -384,55 +384,116 @@ export default {
-      const newResponseToUpsertInOrm = Object.entries(
-        responsesFromApi.values
-      ).map(([questionName, newResponse]) => {
-        const componentType =
-          getComponentTypeOfQuestionByDatasetIdAndQuestionName(
-            this.datasetId,
-            questionName
+      const formattedRecordResponsesForOrm = [];
+      if (responsesFromApi.values) {
+        if (Object.keys(responsesFromApi.values).length === 0) {
+          this.inputs.forEach(
+            ({ question: questionName, options: questionOptions }) => {
+              formattedRecordResponsesForOrm.push({
+                id: responsesFromApi.id,
+                question_name: questionName,
+                options: questionOptions,
+                record_id: responsesFromApi.record_id,
+                user_id: responsesFromApi.user_id ?? null,
+              });
+            }
           );
-        let formattedOptions = getOptionsOfQuestionByDatasetIdAndQuestionName(
-          this.datasetId,
-          questionName
-        );
+        } else {
+          Object.entries(responsesFromApi.values).map(
+            ([questionName, newResponse]) => {
+              const componentType =
+                getComponentTypeOfQuestionByDatasetIdAndQuestionName(
+                  this.datasetId,
+                  questionName
+                );
+              let formattedOptions =
+                getOptionsOfQuestionByDatasetIdAndQuestionName(
+                  this.datasetId,
+                  questionName
+                );
+
+              switch (componentType) {
+                case COMPONENT_TYPE.SINGLE_LABEL:
+                case COMPONENT_TYPE.RATING:
+                  formattedOptions = formattedOptions.map((option) => {
+                    if (option.text === newResponse.value) {
+                      return { id: option.id, text: option.text, value: true };
+                    }
+                    return { id: option.id, text: option.text, value: false };
+                  });
+                  break;
+                case COMPONENT_TYPE.FREE_TEXT:
+                  formattedOptions = [
+                    {
+                      id: formattedOptions[0].id,
+                      text: newResponse.value,
+                      value: newResponse.value,
+                    },
+                  ];
+                  break;
+                default:
+                  console.log(`The component type ${componentType} is unknown`);
+                  return;
+              }
 
-        switch (componentType) {
+              formattedRecordResponsesForOrm.push({
+                id: responsesFromApi.id,
+                question_name: questionName,
+                user_id: responsesFromApi.user_id,
+                record_id: responsesFromApi.record_id,
+                options: formattedOptions,
+              });
+            }
+          );
+        }
+      }
+      return formattedRecordResponsesForOrm;
```

Comment:
add comment to describe why we need the if and else (see line 225 and 238 from RecordFeedbackTaskAndQuestionnaiore.content.vue